pkg/util/platformworkloadidentity/platformworkloadidentityrolesbyversion.go

```diff
@@ -85,3 +86,32 @@ func (service *PlatformWorkloadIdentityRolesByVersionService) GetPlatformWorkloa
 	}
 	return platformWorkloadIdentityRolesByRoleName
 }
+
+func GetPlatformWorkloadIdentityMismatchError(oc *api.OpenShiftCluster, platformWorkloadIdentityRolesByRoleName map[string]api.PlatformWorkloadIdentityRole) error {
+	if !oc.UsesWorkloadIdentity() {
+		return fmt.Errorf("GetPlatformWorkloadIdentityMismatchError called for a Cluster Service Principal cluster")
+	}
+	requiredOperatorIdentities := []string{}
```

Comment:
Optinal suggestion. To me this logic (lines 94-114) would make more sense in ` ValidatePlatformWorkloadIdentityProfile`. This function could be used to return a common error string, or removed entirely.

Reply:
The next PR uses the exactly same thing in frontend validation, also this function is used elsewhere whenever an unexpected identity is faced in the entire code. So to maintain the error message consistency pulled the entire thing here.